src/perception/aruco_processing.cpp

```diff
@@ -0,0 +1,136 @@
+#include "aruco_detect.hpp"
+
+#include "filter.hpp"
+
+// TODO: add cache logic to filter out false positives
+
+/**
+ * Detect fiducials from raw image using OpenCV and calculate their screen space centers.
+ * Maintain the immediate buffer - holds all fiducials seen currently on screen.
+ * Later camera space pose information is filled in when we receive point cloud data.
+ *
+ * @param msg
+ */
+void FiducialsNode::imageCallback(sensor_msgs::ImageConstPtr const& msg) {
+    if (!mEnableDetections) return;
+
+    if (mIsVerbose) {
+        ROS_INFO("Got image %d", msg->header.seq);
+    }
+
+    try {
+        mCvPtr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
+
+        // Detect the fiducial vertices in screen space and their respective ids
+        cv::aruco::detectMarkers(mCvPtr->image, mDictionary, mCornersCache, mIdsCache, mDetectorParams);
```

Comment:
Yeah sure, the reason why its called a "cache" is because I wanted to prevent a bunch of heap allocations in the update loop and instead have a persistent member variable. The other option is a static variable inside the function but those are kinda gross